### PR TITLE
Bump liquidity module v1.4.0 based sdk v0.44.0

### PR DESCRIPTION
**Pull request by dongsam**
## Description

- This PR is for https://github.com/cosmos/gaia/pull/942 [start-upgrade](https://github.com/cosmos/gaia/tree/start-upgrade) branch
- Detailed Progress of liquidity module for SDK v0.43.0 and Gaia Upgrade https://github.com/tendermint/liquidity/issues/432

### ~Liquidity module [v1.3.0](https://github.com/tendermint/liquidity/releases/tag/v1.3.0) is released https://github.com/tendermint/liquidity/pull/439~ ( discontinued due to sdk v0.44.0 released )
- [x] Fix some issues related to genesis import/export https://github.com/tendermint/liquidity/pull/437 
- [x] Updating docs
- [x] the same version of [gravity-devs/liquidity v1.3.0](https://github.com/gravity-devs/liquidity/releases/tag/v1.3.0) as well, As was the latest version with Gaia, import to gravity-devs/liquidity, not to tendermint/liquidity.
- [x] Test and simulate https://github.com/tendermint/liquidity/issues/434, https://github.com/b-harvest/gravity-dex-upgrade-test/tree/kogisin/v5.0.5-upgrade-simulation/v5.0.5
 

#### CHANGELOGS

* [\#433](https://github.com/tendermint/liquidity/pull/433) (sdk) Bump SDK version to [v0.43.0](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.43.0).
* [\#436](https://github.com/tendermint/liquidity/pull/436) Validation `MsgSwapWithinBatch` and `OfferCoinFee` ceiling
  * When calculating `OfferCoinFee`, the decimal points are rounded up.
    - before (v1.2.x):  `MsgSwapWithinBatch.OfferCoinFee` should be `OfferCoin` * `params.SwapFeeRate/2` with Truncate or 0
    - after (v1.3.x):  `MsgSwapWithinBatch.OfferCoinFee` should be `OfferCoin` * `params.SwapFeeRate/2` with Ceil
  * Fix reserveOfferCoinFee residual Issue due to decimal error
* [\#438](https://github.com/tendermint/liquidity/pull/438) Fix PoolBatch index, beginHeight issues and genesis logic
  * Remove `PoolBatchIndex`
  * Fix `PoolBatch.Index` duplicated bug
  * Fix `PoolBatch.BeginHeight` consistency issue on genesis init logic  
 

### Liquidity module [v1.4.0](https://github.com/tendermint/liquidity/releases/tag/v1.4.0) is released, [gravity-devs/liquidity v1.4.0](https://github.com/gravity-devs/liquidity/releases/tag/v1.4.0) as well

- [x] Add commit for liquidity module bump [gravity-devs/liquidity v1.4.0](https://github.com/gravity-devs/liquidity/releases/tag/v1.4.0)
- [x] Update CHANGELOG.md

#### CHANGELOGS
* #440 (sdk) Bump SDK version to v0.44.0
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/main/README.md#merging-a-pr))

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]

**Comment by dongsam** [comment]
Cosmos-SDK [v0.44.0](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.44.0) has been released, so will it be upgraded to based v0.44.0 instead of v0.43.0 in Gaia Vega upgrade? Then, we will also prepare for the upgrade of the liquidity module to the based v0.44.0

**Comment by yaruwangway** [comment]
> Cosmos-SDK [v0.44.0](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.44.0) has been released, so will it be upgraded to based v0.44.0 instead of v0.43.0 in Gaia Vega upgrade? Then, we will also prepare for the upgrade of the liquidity module to the based v0.44.0

I think so. Please confirm this @okwme
which IBC-go version will you use ?

**Comment by colin-axner** [comment]
> which IBC-go version will you use ?

[v1.1.0](https://github.com/cosmos/ibc-go/releases/tag/v1.1.0)

**Review by alessio — APPROVED** [review]
Signing off. Thanks, folks :pray:

**Review by jgimeno — APPROVED** [review]

**Comment by dongsam** [comment]
R4R: Since the Cosmos-sdk version of https://github.com/cosmos/gaia/pull/942 was updated based on v0.44.0, the liquidity was also bumped to v1.4.0 release based on sdk v0.44.0 on this PR.
As was the latest version with Gaia, import to `gravity-devs/liquidity`, not to `tendermint/liquidity`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+ * (gaia) Bump [Liquidity](https://github.com/gravity-devs/liquidity) module to [v1.4.0](https://github.com/Gravity-Devs/liquidity/releases/tag/v1.4.0). See the [CHANGELOG.md](https://github.com/Gravity-Devs/liquidity/blob/v1.4.0/CHANGELOG.md#v130---2021-08-31) for details. This contains consensus breaking changes.
```
Comment (alexanderbez) [inline CHANGELOG.md]:
```suggestion
 * (gaia) Bump [Liquidity](https://github.com/gravity-devs/liquidity) module to [v1.4.0](https://github.com/Gravity-Devs/liquidity/releases/tag/v1.4.0). See the [CHANGELOG.md](https://github.com/Gravity-Devs/liquidity/blob/v1.4.0/CHANGELOG.md#v130---2021-08-31) for details. This contains consensus breaking changes.

```

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**